### PR TITLE
Plumbing.remrun

### PR DESCRIPTION
**Pull request by oliver-sanders**
@wxtim 

Highlights:

* Fix the `--host` option to `cylc {run,restart}`.
* Remove `remrun()` for good.
* Remove the associated arguments of `remrun()` (e.g. `--host`).
* Remove the `host`, `owner` and `port` options to `SuiteRuntimeClient` (we should always use the contact file).

Caveats:

* I've not run the full test battery yet...
* The unit test `cylc/flow/tests/network/test_client.py` now fails (but this will be removed in the integration testing PR).

### CODE REVIEW

**Comment by oliver-sanders** [comment]
Three tests failing:

* cylc/flow/tests/network/test_client.py
* cylc/flow/tests/network/test_publisher.py
* tests/hold-release/12-hold-then-retry.t

The first two tests are refactored in the integration test battery so can be ignored (hopefully that branch will be merged soon).

The third test is currently broken on tim/plumbing.

File: cylc/flow/scheduler_cli.py [hunk 1]
```diff
@@ -267,20 +267,19 @@ def scheduler_cli(parser, options, args, is_restart=False):
 
     # Check whether a run host is explicitly specified, else select one.
     if not options.host:
-        host = select_suite_host()[0]
-        if is_remote_host(host):
+        options.host = select_suite_host()[0]
+    if options.host:
```
Comment (hjoliver) [inline cylc/flow/scheduler_cli.py]:
```suggestion
    else:
```

File: cylc/flow/scripts/cylc_check_versions.py [hunk 2]
```diff
@@ -55,96 +53,63 @@ def get_option_parser():
...
-    # but this command only exists for version > 6.3.0.
-    # So for the moment generate an actual remote invocation command string for
-    # "cylc --version".
-
-    # (save verbose flag as gets reset in remrun)
+
+    platforms = {
+        config.get_config(['runtime', name, 'platform'])
+        for name in config.get_namespace_list('all tasks')
+    } ^ {None, 'localhost'}
```
Comment (hjoliver) [inline cylc/flow/scripts/cylc_check_versions.py:77]: Fancy :tada:

**Review by hjoliver — APPROVED** [review]
Cursory approval granted :boom:

File: cylc/flow/scheduler_cli.py (hunk 1 above)
Comment (oliver-sanders) [reply cylc/flow/scheduler_cli.py]: The previous block sets `options.host`.

Comment (hjoliver) [reply cylc/flow/scheduler_cli.py]: Ah, so it does.

**Comment by oliver-sanders** [comment]
Yikes @wxtim don't try to merge the comparison branch into a PR 😈.

I've cherry-picked my commits onto your branch. At least I think these were the commits.

```
$ git log plumbing.remrun.safe --oneline

# these were the remrun commits
c7a817c1a (plumbing.remrun-safe, plumbing.remrun-before) tests: fix syntax
c2e128fef remove --host --owner --port --ssh-cylc --no-login --use-ssh
3373a18dd remote: remove remrun functionality

# these commits should already be on wxtim/platforms
b2034280e scheduler_cli: fix --host re-invocation
f89b4a4cc pycodestyle fix
aa80ddcdf tests: fix tests/cylc-cat-log/{07,08}
936bf7c0b tests: upgrade tests/cylc-cat-log/09
df65818d8 tests: upgrade tests/cylc-cat-log/10
8d4607210 tests: upgrade tests/cylc-cat-log/11
b8b6b4a2c cylc_cat_log: fix $HOME expansion
```

**Comment by oliver-sanders** [comment]
Same old unit-test failure, all other tests passing.

**Pull request merged**